### PR TITLE
Fix samtools wrappers to work with fasta.gz refs

### PR DESCRIPTION
**Pull request by wm75**
Galaxy's fasta.gz datatype is currently used for gzipped and bgzipped
fasta files alike, but samtools faidx can only handle the latter
directly.
Until Galaxy starts supporting a dedicated fasta.bgzip datatype, this
change works around the existing datatype's ambiguity with an
"easier to ask for forgiveness than permission" approach letting
samtools faidx try to index the file, then, if that fails, try again with the
uncompressed (likely regular gzipped) file.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by wm75** [comment]
@bernt-matthias if you like this solution, feel free to borrow it for https://github.com/galaxyproject/tools-iuc/pull/3549 :)

**Comment by wm75** [comment]
The disadvantage of this workaround is that the standard user that passes in a gzip-compressed fasta.gz file will not be alerted to the fact that they are doing something rather inefficient, but at least things will just work for now.
The dedicated fasta.bgzip datatype is the cleaner long-term solution though, imo.

File: tool_collections/samtools/macros.xml [hunk 1]
```diff
@@ -50,23 +50,56 @@
...
+                        samtools faidx \$reffa;
+                    }
+                } &&
+            #end if
+            reffai=\$reffa.fai &&
         #elif $addref_cond.addref_select == "cached":
-            #set reffa=str($addref_cond.ref.fields.path)
-            #set reffai=str($addref_cond.ref.fields.path)+".fai"
+            ## in case of cached the absolute path is used which allows to read 
+            ## a cram file  without specifying the reference
```
Comment (wm75) [inline tool_collections/samtools/macros.xml:98]: I didn't really understand this comment so I decided to keep it, but move it to the relevant section of the code.

**Comment by wm75** [comment]
Very weird linting problem:

https://github.com/galaxyproject/tools-iuc/runs/4191216022?check_suite_focus=true#step:5:346

why does it think there is an optional="false" combined with checkboxes, when that is not present in the xmls?

**Comment by bernt-matthias** [comment]
> Very weird linting problem:
> 
> https://github.com/galaxyproject/tools-iuc/runs/4191216022?check_suite_focus=true#step:5:346
> 
> why does it think there is an optional="false" combined with checkboxes, when that is not present in the xmls?

Indeed. Its a minor bug in the linter: https://github.com/galaxyproject/galaxy/pull/12920

**Comment by bgruening** [comment]
And lets retrigger the tests.

**Review by bernt-matthias — APPROVED** [review]

**Comment by bernt-matthias** [comment]
https://github.com/galaxyproject/galaxy/issues/12995

**Pull request merged**